sycl/include/CL/sycl/access/access.hpp

```diff
@@ -127,7 +127,15 @@ struct PtrValueType<ElementType, access::address_space::global_space> {
 
 template <typename ElementType>
 struct PtrValueType<ElementType, access::address_space::constant_space> {
-  using type = SYCL_CONSTANT_AS ElementType;
+// Current implementation of address spaces handling leads to possibility
+// of emitting incorrect (in terms of OpenCL) address space casts from constant
+// to generic (and vise-versa). So, global address space is used here instead of
+// constant to avoid incorrect address space casts in the produced device code.
+// "const" qualifier is not used here because multi_ptr interface contains
+// methods which return pure ElementType without qualifiers and adding const
```

Comment:
I'm not sure I follow this comment.
Can we define `type` alias as `using type = const SYCL_GLOBAL_AS ElementType;`?

Reply:
Do you think there is something to improve in the specification?

Reply:
> I'm not sure I follow this comment.
> Can we define `type` alias as `using type = const SYCL_GLOBAL_AS ElementType;`?

No. I didn't add the const qualifier here because multi_ptr interface contains methods and operators which take/return pure ElementType parameter without qualifiers and address spaces. If I use the const qualifier here, I will be forced to add const_casts to these methods to remove const qualifier from underlying pointer type. So adding const qualifier here seems unnecessary.